From 890c2bcddfd6abf73873fe7763ff4456d30c84fa Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Tue, 14 Aug 2018 15:10:52 -0700
Subject: [PATCH] Auto fix bad Y levels on player login

Bring down to a saner Y level if super high, as this can cause the server to crash

diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
index e0e2b993..09b07f7c 100644
--- a/src/main/java/net/minecraft/server/EntityPlayer.java
+++ b/src/main/java/net/minecraft/server/EntityPlayer.java
@@ -123,6 +123,7 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
 
     public void a(NBTTagCompound nbttagcompound) {
         super.a(nbttagcompound);
+        if (this.locY > 300) this.locY = 257; // Paper - bring down to a saner Y level if out of world
         if (nbttagcompound.hasKeyOfType("playerGameType", 99)) {
             if (MinecraftServer.getServer().getForceGamemode()) {
                 this.playerInteractManager.setGameMode(MinecraftServer.getServer().getGamemode());
-- 
2.50.1

